### PR TITLE
LPS-192911

### PR DESCRIPTION
**Pull request by icgsbr**
https://liferay.atlassian.net/browse/LPS-192911

### CODE REVIEW

**Comment by icgsbr** [comment]
ci:test:relevant

**Comment by icgsbr** [comment]
ci:test:forms

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-evaluator-impl/src/main/java/com/liferay/dynamic/data/mapping/form/evaluator/internal/helper/DDMFormEvaluatorHelper.java [hunk 1]
```diff
@@ -125,13 +128,45 @@ public DDMFormEvaluatorEvaluateResponse evaluate() {
...
+
 		_validateFields();
 
 		_localizeNumericDDMFormFieldValues();
 
 		return _buildDDMFormEvaluatorEvaluateResponse();
 	}
 
+	protected void changeRequiredFieldsOnDisabledPages() {
+		if (getDisabledPagesIndexes().isEmpty() || (_ddmFormLayout == null)) {
```
[the remark this answers is not in the document]
Comment (carolmariaabb) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-evaluator-impl/src/main/java/com/liferay/dynamic/data/mapping/form/evaluator/internal/helper/DDMFormEvaluatorHelper.java:141]: @icgsbr use `SetUtil.isEmpty(getDisabledPagesIndexes())` instead

**Review by carolmariaabb — CHANGES_REQUESTED** [review]

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-evaluator-impl/src/main/java/com/liferay/dynamic/data/mapping/form/evaluator/internal/helper/DDMFormEvaluatorHelper.java [hunk 2]
```diff
@@ -125,13 +128,45 @@ public DDMFormEvaluatorEvaluateResponse evaluate() {
...
+			return;
+		}
+
+		for (Object index : getDisabledPagesIndexes().toArray()) {
```
[the remark this answers is not in the document]
Comment (icgsbr) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-evaluator-impl/src/main/java/com/liferay/dynamic/data/mapping/form/evaluator/internal/helper/DDMFormEvaluatorHelper.java:145]: On it!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-evaluator-impl/src/main/java/com/liferay/dynamic/data/mapping/form/evaluator/internal/helper/DDMFormEvaluatorHelper.java [hunk 3]
```diff
@@ -125,13 +128,45 @@ public DDMFormEvaluatorEvaluateResponse evaluate() {
...
+			DDMFormLayoutPage page = _ddmFormLayout.getDDMFormLayoutPage(
```
[the remark this answers is not in the document]
Comment (icgsbr) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-evaluator-impl/src/main/java/com/liferay/dynamic/data/mapping/form/evaluator/internal/helper/DDMFormEvaluatorHelper.java:146]: On it!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-evaluator-impl/src/main/java/com/liferay/dynamic/data/mapping/form/evaluator/internal/helper/DDMFormEvaluatorHelper.java [hunk 4]
```diff
@@ -125,13 +128,45 @@ public DDMFormEvaluatorEvaluateResponse evaluate() {
...
+				(Integer)index);
+
+			for (DDMFormLayoutRow rows : page.getDDMFormLayoutRows()) {
```
[the remark this answers is not in the document]
Comment (icgsbr) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-evaluator-impl/src/main/java/com/liferay/dynamic/data/mapping/form/evaluator/internal/helper/DDMFormEvaluatorHelper.java:149]: On it!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-evaluator-impl/src/main/java/com/liferay/dynamic/data/mapping/form/evaluator/internal/helper/DDMFormEvaluatorHelper.java [hunk 5]
```diff
@@ -125,13 +128,45 @@ public DDMFormEvaluatorEvaluateResponse evaluate() {
...
+				for (DDMFormLayoutColumn columns :
+						rows.getDDMFormLayoutColumns()) {
```
[the remark this answers is not in the document]
Comment (icgsbr) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-evaluator-impl/src/main/java/com/liferay/dynamic/data/mapping/form/evaluator/internal/helper/DDMFormEvaluatorHelper.java:151]: On it!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-evaluator-impl/src/main/java/com/liferay/dynamic/data/mapping/form/evaluator/internal/helper/DDMFormEvaluatorHelper.java [hunk 6]
```diff
@@ -125,13 +128,45 @@ public DDMFormEvaluatorEvaluateResponse evaluate() {
...
+						_ddmFormFieldsMap.get(
+							fieldName
+						).setRequired(
+							false
+						);
+						_ddmFormFieldsMap.get(
+							fieldName
+						).setDDMFormFieldValidation(
+							null
+						);
```
[the remark this answers is not in the document]
Comment (icgsbr) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-evaluator-impl/src/main/java/com/liferay/dynamic/data/mapping/form/evaluator/internal/helper/DDMFormEvaluatorHelper.java:163]: On it!

[the remark this answers is not in the document]
Comment (icgsbr) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-evaluator-impl/src/main/java/com/liferay/dynamic/data/mapping/form/evaluator/internal/helper/DDMFormEvaluatorHelper.java]: Done!

**Comment by icgsbr** [comment]
ci:test:sf

**Comment by icgsbr** [comment]
ci:test:stable

**Comment by carolmariaabb** [comment]
ci:test:forms

**Comment by marcelabc** [comment]
ci:reevaluate:1397401_6574

**Comment by carolmariaabb** [comment]
ci:forward:force

**Comment by icgsbr** [comment]
Hi, for what I've noticed, the error found on relevant is not related with the alterations made on this PR.

**Pull request closed without merging**